app/controllers/registration_controller.rb

```diff
@@ -22,13 +22,23 @@ def new
   end
 
   def destroy
-    @registration = Registration.find(params[:registration_id])
-    if current_user == @registration.user && @registration.event.start_date.future?
+    is_registrationId_given = params[:registration_id].present?
+    if is_registrationId_given
+      @registration = Registration.find(params[:registration_id])
+    elsif current_user.is_admin?
```

Comment:
Why do you need to check if `current_user` is admin or not?

Reply:
I want to verify admin in order to delete. Only the admin could see the list of students. I guess we do not need to verify it again.

Reply:
Exactly!